packages/gatsby/src/schema/infer/is-file.ts

```diff
@@ -0,0 +1,102 @@
+import path from "path"
+import { slash } from "gatsby-core-utils"
+import mime from "mime"
+import isRelative from "is-relative"
+import isRelativeUrl from "is-relative-url"
+import { getValueAt } from "../../utils/get-value-at"
+import { Node } from "../../../index"
+import { INodeStore } from "../../db/types"
+
+const getFirstValueAt = (node: Node, selector: string | string[]): any => {
+  let value = getValueAt(node, selector)
+  while (Array.isArray(value)) {
+    value = value[0]
+  }
+  return value
+}
+
+const findAncestorNode = (
+  nodeStore: INodeStore,
+  childNode: Node,
+  predicate: (n: Node) => boolean
+): Node | null => {
+  let node: Node | undefined = childNode
+  do {
+    if (predicate(node)) {
+      return node
+    }
+  } while ((node = node.parent ? nodeStore.getNode(node.parent) : undefined))
+  return null
+}
+
+const getBaseDir = (nodeStore: INodeStore, node: Node): string | null => {
+  if (node) {
+    const { dir } = findAncestorNode(
+      nodeStore,
+      node,
+      node => node.internal.type === `File`
+    ) || { dir: `` }
+
+    return typeof dir === `string` ? dir : null
+  }
+  return null
+}
+
+const withBaseDir = (dir: string) => (p: string): string =>
+  path.posix.join(dir, slash(p))
+
+const getAbsolutePath = (
+  nodeStore: INodeStore,
+  node: Node,
+  relativePath: string
+): string | string[] | null => {
+  const dir = getBaseDir(nodeStore, node)
+  const withDir = withBaseDir(dir ?? ``)
+  return dir
+    ? Array.isArray(relativePath)
+      ? relativePath.map(withDir)
+      : withDir(relativePath)
+    : null
```

Comment:
I don't think this check is relevant anymore, possibly ever. `getAbsolutePath` is called from `getFilePath`, and the `relativePath` parameter is a value return from `slash` in `gatsby-core-utils` which is defined as returning a `string`. As such, I think that `Array.isArray` will always return `false`.

Reply:
Looking at some test coverage data for the original JS file suggests that if it is possible to be an array, it's at least untested.

<img width="612" alt="Screenshot 2020-04-17 at 08 31 35" src="https://user-images.githubusercontent.com/1522832/79543753-dcb3d480-8085-11ea-8926-b93bd9f9d699.png">

Reply:
I think you're right. Let's try not to make any more semantic changes in TS convert PR's than we have to. Then in a followup diff refactor the code because it looks to me like this file can use a little cleanup and modernization.